lua/lspconfig.lua

```diff
@@ -18,6 +19,98 @@ end
+        local clients = vim.lsp.get_active_clients()
+        local buffer_filetype = vim.bo.filetype
+        local buffer_dir = vim.fn.expand('%:p:h')
+
+        local win_info = lspui.percentage_range_window(0.8, 0.7)
+        local bufnr, win_id = win_info.bufnr, win_info.win_id
+
+        local buf_lines = {}
+        local header = {
+          "Available servers:",
+          "\t"..table.concat(vim.tbl_keys(configs), ', '),
+          "",
+          "Clients attached to this buffer: "..tostring(#buf_clients)
+        }
+        vim.list_extend(buf_lines, header)
+
+        for _, client in ipairs(buf_clients) do
+          local client_info = {
+            "",
+            "Client: "..tostring(client.id),
+            "\tname: "..client.name,
+            "\troot: "..client.workspaceFolders[1].name,
+            "\tfiletypes: "..table.concat(client.config.filetypes, ', '),
+            "\tcmd: "..table.concat(client.config.cmd, ', '),
+          }
+           vim.list_extend(buf_lines, client_info)
+        end
+
+        local active_section_header = {
+          "",
+          "Total active clients: "..tostring(#clients),
+        }
+        vim.list_extend(buf_lines, active_section_header)
+        for _, client in ipairs(clients) do
+          local client_info = {
+            "",
+            "Client: "..tostring(client.id),
+            "\tname: "..client.name,
+            "\troot: "..client.workspaceFolders[1].name,
+            "\tfiletypes: "..table.concat(client.config.filetypes, ', '),
+            "\tcmd: "..table.concat(client.config.cmd, ', '),
+          }
+           vim.list_extend(buf_lines, client_info)
+        end
+        local matching_config_header = {
+          "",
+          "Clients that match the current buffer filetype:",
+        }
+        vim.list_extend(buf_lines, matching_config_header)
+        for _, config in pairs(configs) do
+          local config_table = config.make_config(buffer_dir)
+
+          local cmd_is_executable, cmd
+          if config_table.cmd then
+            cmd = table.concat(config_table.cmd, " ")
+            if vim.fn.executable(config_table.cmd[1]) == 1 then
+              cmd_is_executable = "True"
+            else
+              cmd_is_executable = "False. Please check your path and ensure the server is installed"
+            end
+          else
+            cmd = "cmd not defined"
+            cmd_is_executable = cmd
+          end
+
+          for _, filetype_match in ipairs(config_table.filetypes) do
+            if buffer_filetype == filetype_match then
+              local matching_config_info = {
+                "",
+                "Config: "..config.name,
+                "\tcmd: "..cmd,
+                "\tcmd is executable: ".. cmd_is_executable,
+                "\tidentified root: "..(config_table.root_dir or "None"),
+                "\tcustom handlers: "..table.concat(vim.tbl_keys(config_table.handlers), ", "),
+              }
```

Comment:
you may be able to re-use some of the `health#*` APIs from checkhealth to do a few of these things if you want. idk if that's better or not.